### PR TITLE
Homework11: Mariya Simonenko

### PR DESCRIPTION
**Pull request by Mary-si**

### CODE REVIEW

**Comment by Mary-si** [comment]
можно уточнить как правильно указать "flag"? чтобы pylint не ругался?

**Comment by Mary-si** [comment]
поправила

File: homework11_library.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    """Инфорация о книгах"""
+
+    # создаем методы класса
+    def __init__(self, title, author, pages, isbn):
+        self.title = title
+        self.author = author
+        self.pages = pages
+        self.isbn = isbn
+        self.status = True
+        # True - доступна. False - взята. None - зарезервирована
```
Comment (Stml89) [inline homework11_library.py:23]: a bit complex. It is better to introduce 1 more flag: is_reserved to show reserved status

```
self.is_available = True
self.is_reserved = True
```

File: homework11_library.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    # def check_status_book(self):
```
Comment (Stml89) [inline homework11_library.py:25]: if you will not use it - remove

File: homework11_library.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    #     return self.flag
+
+    def take(self):
+        """статус"""
+        if self.status is True:
+            self.status = False
+            return True
+        return False
+
+    def returned(self):
```
Comment (Stml89) [inline homework11_library.py:36]: this method doesn’t return the value, it SET value, better to rename it

File: homework11_library.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        """статус"""
+        self.status = True
+
+    def reservation(self):
+        """статус"""
+        if self.status is True:
```
Comment (Stml89) [inline homework11_library.py:42]: `if self.status:`

File: homework11_library.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            self.reservated_book = book
+            print(f"Пользователь забронировал книгу: {self.name},"
+                  f"{self.surname}, {self.__id_number}")
+
+
+user_1 = User("Mariya", "Simonenko", 1234589)
+user_2 = User("Ivan", "Ivanov", 8944589)
+user_3 = User("Olya", "Ignatenko", 2374549)
+
+# Проверка
```
Comment (Stml89) [inline homework11_library.py:101]: user_1 took a book, but user_3 can return the same book? But user_3 didn't take this book:
```
user_1.take_book(book_1)
user_3.return_book(book_1)
```

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework11_bank_deposit.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+class Deposit:  # pylint: disable=too-few-public-methods
+    """Инфорация депозитов"""
+    # создаем методы класса
+    def __init__(self, deposit_start_date, end_date_deposit):
+        self.deposit_start_date = deposit_start_date
+        self.end_date_deposit = end_date_deposit
+
+
+deposit_1 = Deposit("08.09.2021", "08.09.2025")
```
[the remark this answers is not in the document]
Comment (Mary-si) [reply homework11_bank_deposit.py:24]: поправила

File: homework11_library.py (hunk 5 above)
Comment (Mary-si) [reply homework11_library.py:101]: тут имелось ввиду показать User_3  что книга уже взята
поправила.

File: homework11_library.py (hunk 2 above)
Comment (Mary-si) [reply homework11_library.py:25]: +

File: homework11_library.py (hunk 1 above)
Comment (Mary-si) [reply homework11_library.py:23]: поправила

File: homework11_bank_deposit.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+# (вклад с возможностью ежемесячной капитализации - это означает,
+# что проценты прибавляются к сумме вклада ежемесячно).
+#
+# Написать класс Bank, метод deposit принимает аргументы N и R,
+# и возвращает сумму, которая будет на счету пользователя.
+#
+# https://myfin.by/wiki/term/kapitalizaciya-procentov
+
+
+class Deposit:  # pylint: disable=too-few-public-methods
```
[the remark this answers is not in the document]
Comment (Stml89) [reply homework11_bank_deposit.py]: What about this comment?

**Comment by Stml89** [comment]
There is a conflict, to be able to merge this PR you need to resolve conflict first https://docs.github.com/ru/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github

File: homework11_bank_deposit.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (Mary-si) [reply homework11_bank_deposit.py]: удалила, это пробовала чтобы pylint не ругался

File: homework11_library.py (hunk 4 above)
Comment (Mary-si) [reply homework11_library.py]: поправила

**Comment by Stml89** [comment]
> There is a conflict, to be able to merge this PR you need to resolve conflict first https://docs.github.com/ru/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github

Still need to resole a conflict

**Comment by Mary-si** [comment]
> > There is a conflict, to be able to merge this PR you need to resolve conflict first https://docs.github.com/ru/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github
> 
> Still need to resole a conflict, if need assistance let me know

надо попробовать 


выбрать это?

**Comment by Stml89** [comment]
Давай посмотрим что он скажет на "create a merge commit"

**Comment by Mary-si** [comment]
+

File: .github/.DS_Store
Comment (Stml89) [inline .github/.DS_Store:1]: remove this file from the repo, should be in the `.gitignore`

File: .DS_Store
Comment (Stml89) [inline .DS_Store:1]: remove this file from the repo, should be in the `.gitignore`

**Review by Stml89 — CHANGES_REQUESTED** [review]

**Comment by Mary-si** [comment]
поправила

**Comment by Stml89** [comment]
> поправила

I still can see a lot of extra files and directories in the repository

**Comment by Mary-si** [comment]
> > поправила
> 
> I still can see a lot of extra files and directories in the repository

не так должно быть?
Снимок экрана 2024-07-10 в 19 07 56

**Comment by Mary-si** [comment]
добавила 
Снимок экрана 2024-07-15 в 00 18 06

**Comment by Stml89** [comment]
Привет, репозиторий должен быть чистым, а это значит, что:
надо удалить:
1) .github/.DS_Store
2) Папку .venv
3) .DS_Store
4) Папку .idea

File: .idea/.DS_Store
Comment (Stml89) [inline .idea/.DS_Store:1]: remove this file

File: .github/.DS_Store
Comment (Stml89) [inline .github/.DS_Store:1]: remove this file

File: .DS_Store
Comment (Stml89) [inline .DS_Store:1]: remove this file

File: .gitignore
Comment (Stml89) [inline .gitignore:1]: pls, restore this file. it tracks what should be in the repo and wise versa

**Review by Stml89 — CHANGES_REQUESTED** [review]
what is the differences between
1) `homework11_library1.py` and homework11_library.py?
2) `homework11_bank_deposits.py` and `homework11_bank_deposit.py`

**Comment by Mary-si** [comment]
вроде все проверь пожалуйста

File: .github/.DS_Store
Comment (Stml89) [inline .github/.DS_Store:1]: Move it to `.gitignore`

File: .DS_Store
Comment (Stml89) [inline .DS_Store:1]: Move it to `.gitignore`

**Review by Stml89 — CHANGES_REQUESTED** [review]
Overall, HW looks good, but you need to fix the comments and remove the `.idea` directory. I marked it with 5, but when you fix the comments, I'll revise the grade.

File: .DS_Store
Comment (Mary-si) [reply .DS_Store:1]: я вроде сделала, но не понимаю

**Review by Stml89 — APPROVED** [review]

**Pull request merged**